macros/vc.asm

```diff
@@ -0,0 +1,15 @@
+vc_hook: MACRO
+    if DEF(_CRYSTALVC)
+        .VC_\1::
+	if (strcmp("BiographySave_ret", "\1") == 0)
+		assert .VC_BiographySave_ret == $4B84, "vc_hook BiographySave_ret is not at address $4B84.\n This hook is hardcoded in the Virtual Console.\n Fix the address or remove it from your code and pokecrystalvc.patch.template"
+		assert BANK(@) == $5, "vc_hook BiographySave_ret is not in bank $05.\n This hook is hardcoded in the Virtual Console.\n Fix the address or remove it from your code and pokecrystalvc.patch.template"
+	endc
```

Comment:
This is wrong. The address of the `BibliographySave_ret` patch isn't hardcoded. You can see an `Address = {{Address}}` relating to this in the patch file.
What is hardcoded, however, is the operation the Mode = 2; Type = 60 hook performs. Since the address of sMobileEventIndex and sMobileEventIndexBackup are nowhere to be found in the patch file, and these are the addresses it modifies to enable the event, I'm assuming these two are hardcoded. (I'm not sure if both are edited, or whether it's just one of them. If you feel up to it, make the hook execute at bootup (so `Address = 0x150`), boot the ROM with a blank save file, and see what happens.)

Please make a `vc_assert` macro that only executes if `DEF(_CRYSTALVC)`, as in:
```
vc_assert: MACRO
if DEF(_CRYSTALVC)
    assert \1, \2
endc
ENDM
```
And execute this macro right next to the `vc_hook BibliographySave_ret`.

Reply:
Not sure if this is what you wanted. Didn't know if you are referencing some weird symbol interpolation thing i'm unaware of with the `{,Backup}` thing... Let me know if this is too noisy or you had something else in mind.

[989893b](https://github.com/pret/pokecrystal/pull/882/commits/989893b3527ac8f888557d8e66859e22965b1442)

Reply:
Nah, this is what I had in mind. It's on the noisy side, maybe it can be lessened by using the line continuation feature of recent pokecrystal, but for now I'd leave it as is.